src/ConnectedMode.UnitTests/Migration/ConnectedModeMigrationTests.cs

```diff
@@ -189,21 +191,65 @@ public async Task Migrate_HasProgressListener_ReportsProgress()
             progressMessages.Should().NotBeEmpty();
         }
 
+        [TestMethod]
+        [DataRow(true, false)]
+        [DataRow(false, true)]
+        public async Task Migrate_ConnectToSonarQubeIfNeeded_DisconnectedNotCalled(bool isConnectedToNewConnectedMode, bool expectedResult)
```

Comment:
If we successfully migrate the connection stays as is.

Reply:
This tests the `migration successful` case, doesn't it?

If so:
* it might help to make the method name even longer...
* do we need both parameters? i.e. is the second always the inverse of the first?


e.g.

```
public async Task Migrate_ConnectToSonarQubeIfNeeded_MigrationSucceeds_DisconnectedNotCalled(bool isAlreadyConnected)

```